### PR TITLE
Add release pipeline

### PR DESCRIPTION
**Pull request by inverse**
- Tidy up makefile
- Add release pipeline that will add binaries to release when tag is published - e.g. draft release on GitHub and publish :)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -6,14 +6,14 @@ BUILD_DATE := $(shell date -u '+%Y-%m-%d_%H:%M:%S')
 BUILD_DIR = bin
 
 build-all:
-	GOOS=linux GOARCH=amd64 make build
-	GOOS=linux GOARCH=arm make build
+	make build-arm
+	make build-amd64
 
-build-prod:
-	GOOS=linux GOARCH=arm RELEASE=prod make build
```
Comment (inverse) [inline Makefile:13]: RELEASE feels like it should be more version - like GitHub tag to allow us to understand what it was built against :)

Comment (paulotruta) [reply Makefile:13]: Agree, but we still need a way to set the flavour we want to run here ("cloud hosted" == cloud or "home network self-hosted" == prod). But changing RELEASE to be the tag version sounds good to me :)

File: Makefile [hunk 2]
```diff
@@ -22,7 +22,6 @@ build:
 		-X ${PROJECT}/internal/diagnostics.Commit=${COMMIT} \
 		-X ${PROJECT}/internal/diagnostics.BuildDate=${BUILD_DATE}" \
 		-o bin/edgeboxctl-${GOOS}-${GOARCH} ${PROJECT}/cmd/edgeboxctl
-	cp ./bin/edgeboxctl-${GOOS}-${GOARCH} ./bin/edgeboxctl
```
[the remark this answers is not in the document]
Comment (paulotruta) [reply Makefile:25]: Let's remove it and then deal with the consequences, in the name of a cleaner command 👍

**Review by paulotruta — APPROVED** [review]
Will test it out in the current flow and if all looks good, approve 👍

**Pull request closed without merging**